### PR TITLE
Fixed #1303 - API to remove stored session cookies & auth tokens

### LINKED ISSUE #1303: API to remove stored session cookies & auth tokens
API: https://github.com/couchbaselabs/couchbase-lite-api/issues/156
iOS: https://github.com/couchbase/couchbase-lite-ios/issues/1302

### PR DESCRIPTION
**Pull request by hideki**
- Ported iOS implementation: https://github.com/couchbase/couchbase-lite-ios/pull/1330

### CODE REVIEW

File: src/main/java/com/couchbase/lite/replicator/Replication.java [hunk 1]
```diff
@@ -739,6 +747,23 @@ public void deleteCookie(String name) {
...
+    @InterfaceAudience.Public
+    public boolean removeStoredCredentials(){
+        if (getAuthenticator() != null) {
+            if (!(getAuthenticator() instanceof Authorizer) ||
+                    !((Authorizer) getAuthenticator()).removeStoredCredentials())
+                return false;
+        }else {
+            // CBL Java does not use credential. No thing to do
+        }
+        replicationInternal.resetCookieStore();
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: This should only remove the cookies of the replicator's remote URL.

Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: See: https://github.com/couchbase/couchbase-lite-ios/commit/7863a9ba0a40e2cae63dda73e7903b9182fd1cb4

File: src/main/java/com/couchbase/lite/replicator/Replication.java [hunk 2]
```diff
@@ -694,6 +696,12 @@ public void setFilterParams(Map<String, Object> filterParams) {
         replicationInternal.setFilterParams(filterParams);
     }
 
+    /** The server user name that the authenticator has logged in as, if known. Observable. */
+    @InterfaceAudience.Public
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/Replication.java]: I think Java has no concept of Observable.

File: src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java [hunk 3]
```diff
@@ -104,6 +105,7 @@
     protected String lastSequence;
     protected Authenticator authenticator;
     protected boolean authenticating = false;
+    protected String username;
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java]: Shouldn't it be private? Also getUsername() is package level.

**Comment by hideki** [comment]
Hi @pasin,
Thank you for reviewing PR. I applied your feedbacks. Could you please re-review PR?
Thanks!

**Pull request merged**